spec/integration/parser/compiler_spec.rb

```diff
@@ -153,6 +153,17 @@ class c inherits b { notify { hi: } }
     end
   end
 
+  context 'when working with $settings name space' do
+    include PuppetSpec::Compiler
+    it 'makes $settings::strict available as string' do
+      node = Puppet::Node.new("testing")
+      catalog = compile_to_catalog(<<-MANIFEST, node)
+          notify { 'test': message => $settings::strict == 'warning' }
```

Comment:
i saw the boolean addition.  seems fully tested if the strict setting is a Symbol.  i'm not overly worried about arrays and hashes, do we even have settings of those types?